### PR TITLE
Show more than default 100 errors

### PR DESCRIPTION
**Pull request by pitr-ch**
not to hide actual errors under pile of missing symbols

### CODE REVIEW

**Comment by chrisseaton** [comment]
Is this why we never see DSL errors?

**Comment by eregon** [comment]
I never had this problem, what kind of code trigger this?
Misspelling a method name shows clearly one error:
```
src/main/java/org/truffleruby/core/exception/CoreExceptions.java:50: error: cannot find symbol
                Layouts.EXCEPTION.getLogcalClass(exception),
                                 ^
  symbol:   method getLogcalClass(DynamicObject)
  location: variable EXCEPTION of type ExceptionLayout
1 error

Compiling truffleruby with javac-daemon failed
1 build tasks failed
```

**Comment by chrisseaton** [comment]
What do you see if you have a typo in something like a guard string, @eregon? I see a hundred undefined symbols for classes the DSL is supposed to build, but no actual DSL errors.

**Comment by eregon** [comment]
This seems to happen, if for example I have a DSL error like:
```java
        @TruffleBoundary
        @Specialization
        public boolean vmMethodIsBasic(VirtualFrame 
```

Eclipse would show it clearly for me, so I never had this problem.

Without this PR, we see only ` error: cannot find symbol`, with it we get:
```
... lots of cannot find symbol errors

src/main/java/org/truffleruby/core/VMPrimitiveNodes.java:158: error: Method vmMethodIsBasic cannot be annotated with @TruffleBoundary and have a parameter of type VirtualFrame
        public boolean vmMethodIsBasic(VirtualFrame frame, DynamicObject method) {
                       ^
498 errors
Result = 1
```
So :+1: to this PR then.
I think it's so much more convenient if your IDE can show these errors though.

**Comment by chrisseaton** [comment]
> I think it's so much more convenient if your IDE can show these errors though.

Yes, it would be nice. But it was crazy that even the standard command line build tool couldn't, so thanks a lot to @pitr-ch for finally figuring this out.

File: tool/jt.rb [hunk 1]
```diff
@@ -516,7 +516,9 @@ def build(*options)
       raw_sh "make", "-j#{cores}", chdir: "#{TRUFFLERUBY_DIR}/src/main/c"
     when nil
       mx 'sforceimports'
-      mx 'build', '--force-javac', '--warning-as-error'
+      mx 'build', '--force-javac', '--warning-as-error',
+         # show more than default 100 errors not to hide actual errors under pile of missing symbols
+         '-A-Xmaxerrs', '-A1000'
```
Comment (eregon) [inline tool/jt.rb]: Should we also have this setting in CI?
Or you always `jt build` on the command line and IntelliJ never compiles by itself?

Comment (chrisseaton) [reply tool/jt.rb]: > Or you always jt build on the command line

Yes that's what I do. Usually `mx build run ...`.

**Comment by pitr-ch** [comment]
I've never payed attention that it's exactly 100 bugs, however today it finally started to look suspicious :) `javac` should really be warning that it's omitting errors form the output.

**Pull request merged**